modules/dnn/src/layers/layer_norm.cpp

```diff
@@ -240,11 +261,35 @@ class LayerNormLayerImpl CV_FINAL : public LayerNormLayer
         op->set_input_x_by_name(*last_node, input_tensor_wrapper->name.c_str());
         op->update_input_desc_x(*input_tensor_desc);
         // set inputs : gamma
-        op->set_input_gamma_by_name(*scale_node, scale_tensor_wrapper->name.c_str());
-        op->update_input_desc_gamma(*scale_tensor_desc);
+        if (blobs.empty()) {
+            auto scale_tensor_wrapper = inputs[1].dynamicCast<CannBackendWrapper>();
+            auto scale_tensor_desc = scale_tensor_wrapper->getTensorDesc();
+            auto scale_node = nodes[1].dynamicCast<CannBackendNode>()->getOp();
+            op->set_input_gamma_by_name(*scale_node, scale_tensor_wrapper->name.c_str());
+            op->update_input_desc_gamma(*scale_tensor_desc);
+        } else {
+            const auto &scale_mat = blobs.front();
+            const auto op_const_scale = std::make_shared<CannConstOp>(scale_mat.data, scale_mat.type(), shape(scale_mat), cv::format("%s_w", name.c_str()));
+            op->set_input_gamma(*(op_const_scale->getOp()));
+            op->update_input_desc_gamma(*(op_const_scale->getTensorDesc()));
+        }
         // set inputs : beta
-        op->set_input_beta_by_name(*bias_node, bias_tensor_wrapper->name.c_str());
-        op->update_input_desc_beta(*bias_tensor_desc);
+        if (blobs.empty()) {
```

Comment:
There is `blobs.empty()` check at line 264. I propose to merge the conditions.

Reply:
Resolved